engine/src/main/java/com/google/android/fhir/sync/Sync.kt

```diff
@@ -49,21 +63,23 @@ object Sync {
    * the same [FhirSyncWorker] to retrieve the status of the job.
    *
    * @param retryConfiguration configuration to guide the retry mechanism, or `null` to stop retry.
-   * @return a [Flow] of [SyncJobStatus]
+   * @return a [Flow] of [SyncState]
    */
   inline fun <reified W : FhirSyncWorker> oneTimeSync(
     context: Context,
     retryConfiguration: RetryConfiguration? = defaultRetryConfiguration,
-  ): Flow<SyncJobStatus> {
+  ): Flow<SyncState> {
     val uniqueWorkName = "${W::class.java.name}-oneTimeSync"
     val flow = getWorkerInfo(context, uniqueWorkName)
+    val oneTimeWorkRequest =
+      createOneTimeWorkRequest(retryConfiguration, W::class.java, uniqueWorkName)
     WorkManager.getInstance(context)
       .enqueueUniqueWork(
         uniqueWorkName,
         ExistingWorkPolicy.KEEP,
-        createOneTimeWorkRequest(retryConfiguration, W::class.java),
+        oneTimeWorkRequest,
       )
-    return flow
+    return combineSyncStateForOneTimeSync(context, uniqueWorkName, flow, oneTimeWorkRequest.id)
```

Comment:
we no longer need the id to be passed as the flow we create will contain workInfo state

```suggestion
    return combineSyncStateForOneTimeSync(context, uniqueWorkName, flow)
```

Reply:
done.